src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -906,6 +913,42 @@ private static bool AddFlags(TypeSymbol type, ArrayBuilder<bool> transformFlagsB
 
                 // Continue walking types
                 return false;
+
+                static void handleFunctionPointerType(FunctionPointerTypeSymbol funcPtr, ArrayBuilder<bool> transformFlagsBuilder, bool isNestedNamedType, bool addCustomModifierFlags)
+                {
+                    Func<TypeSymbol, (ArrayBuilder<bool>, bool), bool, bool> visitor = addFlags;
```

Comment:
>visitor = addFlags [](start = 77, length = 18)

This includes an allocation. And I think it would be clearer to inline the local and the local function anyway:
```
twa.Type.VisitType((type, param, isNestedNamedType) =>
    AddFlags(type, param.builder, isNestedNamedType, param.addCustomModifiers),
    (transformFlagsBuilder, addCustomModifierFlags));
```

Reply:
`addFlags` is static, so it should be cached. Personally, I don't believe that it's clearer.

---
In reply to: [422377182](https://github.com/dotnet/roslyn/pull/44078#discussion_r422377182) [](ancestors = 422377182)

Reply:
I don't think there is caching here, even if the local function is static:
[example](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQZgAJ1CBhQgb10JuKJQBZCBZACgEpLraeUM1CAD0IBeQgBsA9gGMI4gNzceNJcsEdFOZbXXtN2mnwBsxJlNniOlQgF9VdnDaA==)

To cache, we would use a static Func field. Here's an example from `TypeSymbolExtensions`:
```
        private static readonly Func<TypeSymbol, TypeParameterSymbol?, bool, bool> s_containsTypeParameterPredicate =
            (type, parameter, unused) => type.TypeKind == TypeKind.TypeParameter && (parameter is null || TypeSymbol.Equals(type, parameter, TypeCompareKind.ConsiderEverything2));

```

---
In reply to: [422416920](https://github.com/dotnet/roslyn/pull/44078#discussion_r422416920) [](ancestors = 422416920,422377182)

Reply:
We could use a `static` field. In this case, there is only one use of the `Func` though so it would be simpler to use a lambda which would be cached automatically.

---
In reply to: [422424873](https://github.com/dotnet/roslyn/pull/44078#discussion_r422424873) [](ancestors = 422424873,422416920,422377182)